### DISCUSSION TITLE
Organization secrets showing up empty in private repo

### DISCUSSION THREAD

**Discussion by tamalsaha** [community/community#25428] [Actions]
<p>Hi,</p>
<p>I am trying to use organization secret in my private organization repo. I am using the new free Team accounts. This secret is a Google cloud service account json file. It is around 2K size. I notice that these secrets are empty in GitHub actions during execution. This seems like a bug. Can you please fix this?</p>
<p>Regards,</p>
<p>Tamal</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/tamalsaha">@tamalsaha</a> , </p>
<p>I am sorry to tell you that using org-level secrets on a free plan, that secrets are not supported for private repos.  So it is not a bug. There should be some indicate in organization secrets tab and repo secrets tab. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cd572621f6d6e11a082734510981ce72211b94fc.png" data-download-href="/uploads/short-url/tiwB1HVaWIgHGjM9Ixk7Fd7nAzq.png?dl=1" title="private secrets.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181091057-ae523e08-b06f-4ba0-be76-e4c1f55d8a81.png" alt="private secrets.png" data-base62-sha1="tiwB1HVaWIgHGjM9Ixk7Fd7nAzq" width="690" height="452" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/cd572621f6d6e11a082734510981ce72211b94fc_2_690x452.png, https://aws1.discourse-cdn.com/github/original/2X/c/cd572621f6d6e11a082734510981ce72211b94fc.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/cd572621f6d6e11a082734510981ce72211b94fc.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cd572621f6d6e11a082734510981ce72211b94fc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">private secrets.png</span><span class="informations">934×612 101 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>If you don’t see this notice in your org, please feel free to let me know. </p>

**Comment by crohr** [comment]
<p>Hello, I have the same problem, and I don’t see the note that you’ve highlighted in your screenshot (e.g. for any repo of the pullpreview organization.</p>

**Comment by bryanherman** [comment]
<p>I also have the same issue. I successfully created a secret at the org level to see all private repositories called ACTION. When I go to print out the secret in the workflow (I am expecting to see ***), I see nothing. When I look at the string length it is 0.</p>
<pre><code>- shell: bash
  env:
    SUPER_SECRET: ${{ secrets.ACTION }}
  run: |
    echo "$SUPER_SECRET"
    echo "${#SUPER_SECRET}"
</code></pre>
<p>I get the result:</p>
<p>echo “$SUPER_SECRET”</p>
<p>echo “${<span class="hashtag">#SUPER_SECRET</span>}”</p>
<p>shell: /bin/bash --noprofile --norc -e -o pipefail {0}</p>
<p>env:<br>
pythonLocation: /opt/hostedtoolcache/Python/3.7.7/x64<br>
SUPER_SECRET:</p>
<p>0</p>
<p>I see no documentation anywhere saying that this is not a free feature for private repos. Can anyone shed some light on this??</p>

**Comment by MarkStega** [comment]
<p>I am having the same issue. My private repository shows the organization level secret, not a disclaimer that they are not available.</p>

**Comment by cschleiden** [comment]
<p>You should see a message like this: </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2c8a7d6a4c98645e429c64819f7aef9aee1c5daa.png" data-download-href="/uploads/short-url/6m1ILaFz4JLn9MdcilkvGCcJek2.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2201819/181091059-b0d70032-52ed-4627-9823-af40d2f417df.png" alt="image" data-base62-sha1="6m1ILaFz4JLn9MdcilkvGCcJek2" width="690" height="137" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/2c8a7d6a4c98645e429c64819f7aef9aee1c5daa_2_690x137.png, https://aws1.discourse-cdn.com/github/original/2X/2/2c8a7d6a4c98645e429c64819f7aef9aee1c5daa.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/2/2c8a7d6a4c98645e429c64819f7aef9aee1c5daa.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2c8a7d6a4c98645e429c64819f7aef9aee1c5daa_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">830×165 45.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> if your organization is on a free plan.

**Comment by MarkStega** [comment]
<p>I don’t see that message, what I see is:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c1511b0940f8eeb8dcc0fc3080cd9ddab38ad994.png" data-download-href="/uploads/short-url/rA9V1zEt3Jk733uu9XMpyUfFnHC.png?dl=1" title="join.me_2020-06-26_15-08-08" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/634561/181091061-e302515e-5de9-4978-b4ef-977d4a76d663.png" alt="join.me_2020-06-26_15-08-08" data-base62-sha1="rA9V1zEt3Jk733uu9XMpyUfFnHC" width="690" height="397" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/c1511b0940f8eeb8dcc0fc3080cd9ddab38ad994_2_690x397.png, https://aws1.discourse-cdn.com/github/optimized/2X/c/c1511b0940f8eeb8dcc0fc3080cd9ddab38ad994_2_1035x595.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/c/c1511b0940f8eeb8dcc0fc3080cd9ddab38ad994.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c1511b0940f8eeb8dcc0fc3080cd9ddab38ad994_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">join.me_2020-06-26_15-08-08</span><span class="informations">1261×726 35.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by cschleiden** [comment]
<p><a class="mention" href="/markstega">@MarkStega</a> Thanks for the screenshot! :bowing_man: Checked some data on our side, and something does not work correctly. For reference, here is what you <em>should</em> see:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/eca8b63849f4dd57ba9516206c38b6c9b86d3f24.png" data-download-href="/uploads/short-url/xLA94H69uUgCQfWM5HCw7KC7eUQ.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2201819/181091064-692eda9f-6dd2-4357-ab14-ad09a6b43adf.png" alt="image" data-base62-sha1="xLA94H69uUgCQfWM5HCw7KC7eUQ" width="690" height="239" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/e/eca8b63849f4dd57ba9516206c38b6c9b86d3f24_2_690x239.png, https://aws1.discourse-cdn.com/github/optimized/2X/e/eca8b63849f4dd57ba9516206c38b6c9b86d3f24_2_1035x358.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/e/eca8b63849f4dd57ba9516206c38b6c9b86d3f24_2_1380x478.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/eca8b63849f4dd57ba9516206c38b6c9b86d3f24_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1882×652 51.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’ll look into this some more, we’ll fix this.</p>

**Comment by adhirajsinghchauhan** [comment]
<p>Our runs started failing recently, and after checking logs, we found out that all secrets are blank. This used to work earlier (i.e. org secrets were passed correctly into private repos), but now it doesn’t.</p>
<p><strong>Why has this behaviour changed, why wasn’t it announced, and why isn’t it mentioned anywhere?</strong><br>
I see what <a class="mention" href="/markstega">@MarkStega</a> sees as well:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cc45aa7aa097b9881085b8fe973ef6204653e0fa.png" data-download-href="/uploads/short-url/t94F4PhikSxOMxbCqHc7hPT4FUm.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/14949823/181091065-eb7070c8-1dfb-42e3-9c0c-2b4a919d961b.png" alt="image" data-base62-sha1="t94F4PhikSxOMxbCqHc7hPT4FUm" width="690" height="174" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cc45aa7aa097b9881085b8fe973ef6204653e0fa_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">920×232 11.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by cschleiden** [comment]
<p>This should be fixed now, you should correctly see a message indicating the private-repository behavior for organizations on a free plan.</p>

**Comment by llucax** [comment]
<p>I was just bitten by this. Even when the option to enable a secret for <strong>all</strong> private repos is disabled, there is an option to pick which repos a secret is for one by one, and private repos are shown there (and selectable). I assumed you still could pick private repos one by one if you can actually do it via the UI.</p>
<p><img src="https://user-images.githubusercontent.com/1031485/181091066-f7a00f50-e3da-46dd-b9b0-e99c720c7181.png" alt="image" data-base62-sha1="Awl42Fas5Wp65CJCiG1mDVMhiY2" width="345" height="352"></p>
<p>But then I also have the empty secret issue. If I go to the repo and define the same secret, I’m even told I’m overriding an organization secret, which I’m not apparently.</p>
<p>(another screenshot was here but apparently I can only embed one media since I’m a new user in the forums)</p>
<p>I understand the limitation but hopefully the UI can be improved to not offer enabling stuff that won’t work at all, it could save time and confusion to users :slight_smile:</p>

**Comment by llucax** [comment]
<p>The other image:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/9/b99b6db88ca7edd3a6e83d1116a2b53e465bdff3.png" data-download-href="/uploads/short-url/qtXkQBOuskehUxoGEB74nmK2bNp.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1031485/181091069-982ead53-a6e1-44d3-906d-3b30b21a11a2.png" alt="image" data-base62-sha1="qtXkQBOuskehUxoGEB74nmK2bNp" width="690" height="110" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/9/b99b6db88ca7edd3a6e83d1116a2b53e465bdff3_2_690x110.png, https://aws1.discourse-cdn.com/github/original/3X/b/9/b99b6db88ca7edd3a6e83d1116a2b53e465bdff3.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/9/b99b6db88ca7edd3a6e83d1116a2b53e465bdff3.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/9/b99b6db88ca7edd3a6e83d1116a2b53e465bdff3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1009×162 11.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jonvaljonathan** [comment]
What plan is needed to have organizational level secrets on a private repo?

**Comment by andrewjhanlon-dmg** [comment]
A possible reason why you can't see Organization-level secrets is possibly because the Org secret was only allowed to be visible to certain repositories. You can make them available to selected repositories, or the organization as a whole:
![image](https://user-images.githubusercontent.com/113203994/219504564-f24535c2-0514-4e00-be51-6452dfead1a4.png)

**Comment by nerder** [comment]
This is also true for environments secrets

**Comment by yabinboxes** [comment]
it's a little confusing paying for each organization subscription by separate instead of enabled all at once

**Comment by Krishnacapcons638** [comment]
Hey,

We are on the Team plan, but my GitHub Actions workflows are still unable to access the organization-level secrets. I have already granted repository access to the private repositories, so these secrets should be available to them.

Do I need to configure anything else for the repositories or organization settings to make the secrets accessible?
